### PR TITLE
ci: add build workflow

### PR DESCRIPTION
**Pull request by PalmDevs**
This pull request removes the current **PR Build** workflow and replaces it with the more generic **Build** workflow which can be manually dispatched in the Actions page or be triggered via a PR review whose review body contains `!build`.

### CODE REVIEW

**Comment by PalmDevs** [comment]
I am aware that the `if` expressions are very unreadable, but GitHub Actions cannot parse multi-line if expressions.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  pull_request_review:
+    types: [submitted, edited]
+  workflow_dispatch:
+
+# * If Expressions (copy-paste)
+
+# * PULL_REQUEST_REVIEW_IF_EXPRESSION: github.event_name == 'pull_request_review' && contains(fromJson('["changes_requested","approved"]'), github.event.review.state) && contains(fromJson('["collaborator","member","owner"]'), github.event.review.author_association) && endsWith(github.event.review.body, '!build')
+# * SHORT_PULL_REQUEST_REVIEW_IF_EXPRESSION: github.event_name == 'pull_request_review'
+# * BUILD_IF_EXPRESSION: (github.event_name == 'pull_request_review' && contains(fromJson('["changes_requested","approved"]'), github.event.review.state) && contains(fromJson('["collaborator","member","owner"]'), github.event.review.author_association) && endsWith(github.event.review.body, '!build')) || (github.event_name == 'workflow_dispatch')
+
```
Comment (oSumAtrIX) [inline .github/workflows/build.yml:13]:
```suggestion
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+jobs:
+  build: 
+    name: Build
+    runs-on: ubuntu-latest
+    # If this pull request is approved, reviewer is a member, a collaborator, or an owner, and review body contains `!build`
+    # or if it's manually dispatched in the Actions tab
+    # * Expression of: BUILD_IF_EXPRESSION
```
Comment (oSumAtrIX) [inline .github/workflows/build.yml:20]:
```suggestion
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    if: ${{ (github.event_name == 'pull_request_review' && contains(fromJson('["changes_requested","approved"]'), github.event.review.state) && contains(fromJson('["collaborator","member","owner"]'), github.event.review.author_association) && endsWith(github.event.review.body, '!build')) || (github.event_name == 'workflow_dispatch') }}
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+      # This condition should be the same as the one in the job
+      # * Expression of: BUILD_IF_EXPRESSION
```
Comment (oSumAtrIX) [inline .github/workflows/build.yml:25]:
```suggestion
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      cancel-in-progress: ${{ (github.event_name == 'pull_request_review' && contains(fromJson('["changes_requested","approved"]'), github.event.review.state) && contains(fromJson('["collaborator","member","owner"]'), github.event.review.author_association) && endsWith(github.event.review.body, '!build')) || (github.event_name == 'workflow_dispatch') }}
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # Make sure the release step uses its own credentials:
+          # https://github.com/cycjimmy/semantic-release-action#private-packages
```
Comment (oSumAtrIX) [inline .github/workflows/build.yml:34]:
```suggestion
```

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: flutter build apk
+      - name: Upload build
+        uses: actions/upload-artifact@v3
+        with:
+          name: revanced-manager
+          path: build/app/outputs/flutter-apk/app-release.apk
+      - name: Notify reviewer that run finished
+        uses: peter-evans/create-or-update-comment@v3
+        # * Expression of: SHORT_PULL_REQUEST_REVIEW_IF_EXPRESSION
```
Comment (oSumAtrIX) [inline .github/workflows/build.yml:63]:
```suggestion
```

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,80 @@
+name: Build
```
Comment (oSumAtrIX) [inline .github/workflows/build.yml:1]: Reflect this change to the workflow file as well:

```suggestion
name: Build PR

```

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -0,0 +1,80 @@
+name: Build
+
+on:
+  pull_request_review:
+    types: [submitted, edited]
```
Comment (oSumAtrIX) [inline .github/workflows/build.yml:5]: Is `edited` required?

```suggestion
    types: [submitted]
```

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
Quite a lot of duplicate code at first sight. Maybe use variables at the top of the workflow file for commands to clean up a bit?

File: .github/workflows/build.yml (hunk 6 above)
Comment (PalmDevs) [reply .github/workflows/build.yml:1]: This workflow can be manually dispatched as well. **Build** fits the name.

File: .github/workflows/build.yml (hunk 7 above)
Comment (PalmDevs) [reply .github/workflows/build.yml:5]: Probably not.

**Pull request merged**